lib/components/RunFrameWithApi/RunFrameWithApi.tsx

```diff
@@ -126,18 +130,39 @@ export const RunFrameWithApi = (props: RunFrameWithApiProps) => {
 
         const manualEditsFile = fsMap.get(manualEditsFilePath)
 
-        const updatedManualEdits = applyPcbEditEventsToManualEditsFile({
-          circuitJson: circuitJson!,
-          editEvents: [ee],
-          manualEditsFile: JSON.parse(manualEditsFile ?? "{}"),
-        })
+        let currentState: ManualEditsFile = {
+          pcb_placements: [],
+          schematic_placements: [],
+          manual_trace_hints: [],
+        }
+        try {
+          if (manualEditsFile) {
+            currentState = JSON.parse(manualEditsFile)
+          }
+        } catch (e) {
+          console.error("Error parsing manual edits file:", e)
+        }
+
+        if (ee?.edit_event_type === "edit_pcb_component_location") {
+          currentState = applyPcbEditEventsToManualEditsFile({
+            circuitJson: circuitJson!,
+            editEvents: [ee],
+            manualEditsFile: currentState,
+          })
+        } else {
+          currentState = applySchematicEditEventsToManualEditsFile({
+            circuitJson: circuitJson!,
+            editEvents: [ee],
+            manualEditsFile: currentState,
+          })
+        }
```

Comment:
just a note, we're still doing too much logic here, core should have a method `applyEditEventsToManualEditsFile`, it doesn't matter whether it's PCB or Schematic